module/dice/d20-roll.mjs

```diff
@@ -180,13 +185,14 @@ export default class D20Roll extends Roll {
    * @param {number} [data.defaultAction]     The button marked as default
    * @param {boolean} [data.chooseModifier]   Choose which ability modifier should be applied to the roll?
    * @param {string} [data.defaultAbility]    For tool rolls, the default ability modifier applied to the roll
+   * @param {string} [data.skillId]           For skill rolls, the skillId of the skill that is being rolled
    * @param {string} [data.template]          A custom path to an HTML template to use instead of the default
    * @param {object} options                  Additional Dialog customization options
    * @returns {Promise<D20Roll|null>}         A resulting D20Roll object constructed with the dialog, or null if the
    *                                          dialog was closed
    */
   async configureDialog({title, defaultRollMode, defaultAction=D20Roll.ADV_MODE.NORMAL, chooseModifier=false,
-    defaultAbility, template}={}, options={}) {
+    defaultAbility, skillId, template}={}, options={}) {
```

Comment:
I'm perhaps not following this correctly. We've added an `options.minimum` to the `D20Roll` class so that it can be applied as part of `D20Roll#configureModifiers`. Why do we also need to pass the `skillId` through `configureDialog` and `_onDialogSubmit`? Is it not enough to configure a `minimum` that we pass to `d20Roll` inside `Actor5e#rollSkill`, like we do with `Actor5e#rollToolCheck`?

Reply:
This is specifially to handle the special case when we use a different ability for a skill check, that was recently added.

Reply:
OK, I think I understand now: The `skillMinimum` that we derive in `Actor5e#rollSkill` is based on the skill minimum and the ability check minimum. Because the ability used for the skill might change after the user has confirmed the roll dialog, we need to re-derive the minimum in `_onDialogSubmit` based on the new ability, and we need to re-call the skill again to perform that calculation.

I was confused because the `skillId` never changes from `rollSkill` to `_onDialogSubmit` but we still need it to perform the calculation again.

This is actually a similar problem to #1650 where we have a complicated rolling pipeline due to the fact that the user may change the ability score during the rolling process.